yt/frontends/enzo_p/data_structures.py

```diff
@@ -0,0 +1,386 @@
+"""
+Data structures for Enzo-P
+
+
+
+"""
+
+#-----------------------------------------------------------------------------
+# Copyright (c) 2017, yt Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+from yt.utilities.on_demand_imports import \
+    _h5py as h5py
+import numpy as np
+import os
+import stat
+
+from yt.data_objects.grid_patch import \
+    AMRGridPatch
+from yt.data_objects.static_output import \
+    Dataset
+from yt.fields.field_info_container import \
+    NullFunc
+from yt.funcs import \
+    ensure_tuple, \
+    get_pbar, \
+    setdefaultattr
+from yt.geometry.grid_geometry_handler import \
+    GridIndex
+from yt.utilities.logger import \
+    ytLogger as mylog
+
+from yt.frontends.enzo_p.fields import \
+    EnzoPFieldInfo
+from yt.frontends.enzo_p.misc import \
+    get_block_info, \
+    get_child_index, \
+    get_root_blocks, \
+    get_root_block_id, \
+    is_parent
+
+class EnzoPGrid(AMRGridPatch):
+    """
+    Class representing a single EnzoP Grid instance.
+    """
+
+    _id_offset = 0
+    _refine_by = 2
+
+    def __init__(self, id, index, block_name, filename = None):
+        """
+        Returns an instance of EnzoPGrid with *id*, associated with
+        *filename* and *index*.
+        """
+        # All of the field parameters will be passed to us as needed.
+        AMRGridPatch.__init__(self, id, filename=filename, index=index)
+        self.block_name = block_name
+        self._children_ids = None
+        self._parent_id = -1
+        self.Level = -1
+
+    def __repr__(self):
+        return "EnzoPGrid_%04d" % self.id
+
+    def get_parent_id(self, desc_block_name):
+        if self.block_name == desc_block_name:
+            raise RuntimeError("Child and parent are the same!")
+        dim = self.index.ds.dimensionality
+        d_block = desc_block_name[1:].replace(":", "")
+        parent = self
+
+        while True:
+            a_block = parent.block_name[1:].replace(":", "")
+            gengap = (len(d_block) - len(a_block)) / dim
+            if gengap <= 1:
+                return parent.id
```

Comment:
would it be possible to change this loop from `while True` to `while gengap > 1`, and then `return parent.id` outside the loop?